src/main/kotlin/gg/skytils/skytilsmod/features/impl/misc/PartyAddons.kt

```diff
@@ -0,0 +1,228 @@
+            awaitingMessage = 2
+        }
+    }
+
+    @SubscribeEvent
+    fun onChat(event: ClientChatReceivedEvent) {
+        if (!Utils.inSkyblock || event.type == 2.toByte() || !Skytils.config.partyAddons) return
+        val message = event.message.formattedText
+
+        if (message == "§f§r" && awaitingMessage != 0) {
+            event.isCanceled = true
+        } else if (message.startsWith("§6Party Members")) {
+            val partyStart = partyStartPattern.find(message)
+            if (partyStart != null) {
+                party.clear()
+                event.isCanceled = true
+            }
+        } else if (message.startsWith("§eParty Leader")) {
+            val leader = leaderPattern.find(message)
+
+            if (leader != null) {
+                val name = leader.groups["name"]?.value ?: return
+                val status = leader.groups["status"]?.value ?: return
+                val rank = leader.groups["rank"]?.value ?: return
+
+                party.add(PartyMember(name, PartyMemberType.LEADER, status, rank))
+                event.isCanceled = true
+            }
+        } else if (message.startsWith("§eParty Moderators: ")) {
+            playerPattern.findAll(message.substringAfter("§eParty Moderators: ")).forEach {
+                val name = it.groups["name"]?.value ?: return
+                val status = it.groups["status"]?.value ?: return
+                val rank = it.groups["rank"]?.value ?: return
+
+                party.add(
+                    PartyMember(
+                        name,
+                        PartyMemberType.MODERATOR,
+                        status,
+                        rank
+                    )
+                )
+            }
+            event.isCanceled = true
+        } else if (message.startsWith("§eParty Members: ")) {
+            playerPattern.findAll(message.substringAfter("§eParty Members: ")).forEach {
+                val name = it.groups["name"]?.value ?: return
+                val status = it.groups["status"]?.value ?: return
+                val rank = it.groups["rank"]?.value ?: return
+
+                party.add(
+                    PartyMember(
+                        name,
+                        PartyMemberType.MEMBER,
+                        status,
+                        rank
+                    )
+                )
+            }
+            event.isCanceled = true
+        } else if (message.startsWith("§cYou are not currently in a party.") && awaitingMessage != 0) {
+            party.clear()
+        } else if (message == "§9§m-----------------------------------------------------§r" && awaitingMessage != 0) {
+            awaitingMessage--
+            if (awaitingMessage == 1 || party.isEmpty()) return
+
+            val component = UMessage("§aParty members (${party.size})\n")
+
+            val player = party.first { it.name == Skytils.mc.thePlayer.name }
```

Comment:
codestyle: 
static import Skytils.mc
pick a different name that's not player (maybe like "me", "self" or something idk)